pkg/resource/deploy/step_generator.go

```diff
@@ -72,8 +72,51 @@ func (sg *stepGenerator) isTargetedUpdate() bool {
 	return sg.updateTargetsOpt != nil || sg.replaceTargetsOpt != nil
 }
 
-func (sg *stepGenerator) isTargetedForUpdate(urn resource.URN) bool {
-	return sg.updateTargetsOpt == nil || sg.updateTargetsOpt[urn]
+// isTargetedForUpdate returns if urn is target for update. The function accommodates
+// `--target-dependents`. `targetDependentsForUpdate` should probably be called if this function
+// returns true.
+func (sg *stepGenerator) isTargetedForUpdate(
+	urn, parent resource.URN, provider string, dependencies []resource.URN) bool {
+	if sg.updateTargetsOpt == nil || sg.updateTargetsOpt[urn] {
+		return true
+	}
+	if provider != "" {
+		res, err := providers.ParseReference(provider)
+		contract.AssertNoError(err)
+		dependencies = append(dependencies, res.URN())
+	}
+	if parent != "" {
+		dependencies = append(dependencies, parent)
+	}
+	for _, dep := range dependencies {
+		if dep != "" && sg.updateTargetsOpt[dep] {
+			return true
+		}
+	}
+	return false
+}
+
+// targetDependentsForUpdate sets the dependents of urn to be updated, if performing a targeted
+// update.
+func (sg *stepGenerator) targetDependentsForUpdate(
```

Comment:
I don't think we need this function. This is adding all of a resource's dependencies to the target set; I think that we just need to add the resource itself.